server/db/db.js

```diff
@@ -1295,6 +1295,37 @@ async function close() {
   }
 }
 
+async function getInactiveGatewaysWithClient(client, pgClient) {
+  try {
+    const results = await helpers.runQuery(
+      'getInactiveGatewaysWithClient',
+      `
+      SELECT *
+      FROM gateways g
+      WHERE g.client_id = $1
+      AND g.sent_vitals_alert_at IS NULL
+      AND g.is_sending_vitals = true
+      LIMIT 1
+      `,
+      [client.id],
+      pool,
+      pgClient,
+    )
+
+    if (results !== undefined && results.rows.length > 0) {
+      return results.rows.map(r => createGatewayFromRow(r, [client]))
```

Comment:
Nice. I like how you passed `client` to the mapping function to avoid unnecessary DB calls.